Makefile

```diff
@@ -0,0 +1,58 @@
+CC := clang++
+C_FLAGS := -g -Wall -Wextra -pedantic -std=c++20 `pkg-config --cflags --libs protobuf` -pthread
+FS_FLAGS := -lfuse3 -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=30
+SERVER_FLAGS := 
+COMMON := 
+SERVER_FILES := 
+FS_FILES :=
+PROTO := proto/packets.proto
+
+all: build
+
+.PHONY: build
+build: filesystem server
+
+.PHONY: clean
+clean: filesystem-clean server-clean
+
+.PHONY: clean-all
+clean-all: clean proto-clean
+
+.PHONY: filesystem-run
+filesystem-run: filesystem 
+	./filesystem/filesystem -f test/
+
+.PHONY: filesystem 
+filesystem: filesystem/filesystem
+
+filesystem/filesystem: proto/proto.pb.o
+	$(CC) $(C_FLAGS) $(FS_FLAGS) -o $@ filesystem/main.cpp $(COMMON) $(FS_FILES) proto/proto.pb.o
+
+.PHONY: filesystem-clean
+filesystem-clean:
+	rm -f filesystem/filesystem
+
+.PHONY: server-run
+server-run: server
+	./server/server ~/name
```

Comment:
może zrobimy jakiś katalog z jakimiś przykładowymi plikami wewnątrz repozytorium, żeby hostować serwerem? możnaby wtedy zrobić jakiś prosty smoke test w CI i też łatwiej chyba będzie tak testować niż zmieniać to ~/name na jakiś swoj przykładowy?

Reply:
+1